perl/p5-dbd-pg/Portfile

```diff
@@ -28,51 +28,32 @@ if {${perl5.major} ne ""} {
+set postgresql_suffixes {10 11 12 13 14 15}
 
-    configure.env           PATH=${prefix}/lib/postgresql10/bin:$env(PATH) \
-                            POSTGRES_INCLUDE=${prefix}/include/postgresql10 \
-                            "POSTGRES_LIB=${prefix}/lib/postgresql10 -lssl"
+set postgresql_ports {}
+foreach s ${postgresql_suffixes} {
+    lappend postgresql_ports postgresql${s}
 }
 
-variant postgresql11 conflicts postgresql10 postgresql12 postgresql13 postgresql14 description {Build with postgresql11 port} {
-    depends_lib-append      port:postgresql11
-
-    configure.env           PATH=${prefix}/lib/postgresql11/bin:$env(PATH) \
-                            POSTGRES_INCLUDE=${prefix}/include/postgresql11 \
-                            "POSTGRES_LIB=${prefix}/lib/postgresql11 -lssl"
-}
-
-variant postgresql12 conflicts postgresql10 postgresql11 postgresql13 postgresql14 description {Build with postgresql12 port} {
-    depends_lib-append      port:postgresql12
-
-    configure.env           PATH=${prefix}/lib/postgresql12/bin:$env(PATH) \
-                            POSTGRES_INCLUDE=${prefix}/include/postgresql12 \
-                            "POSTGRES_LIB=${prefix}/lib/postgresql12 -lssl"
-}
-
-variant postgresql13 conflicts postgresql10 postgresql11 postgresql12 postgresql14 description {Build with postgresql13 port} {
-    depends_lib-append      port:postgresql13
-
-    configure.env           PATH=${prefix}/lib/postgresql13/bin:$env(PATH) \
-                            POSTGRES_INCLUDE=${prefix}/include/postgresql13 \
-                            "POSTGRES_LIB=${prefix}/lib/postgresql13 -lssl"
-}
-
-variant postgresql14 conflicts postgresql10 postgresql11 postgresql12 postgresql13 description {Build with postgresql14 port} {
-    depends_lib-append      port:postgresql14
-
-    configure.env           PATH=${prefix}/lib/postgresql14/bin:$env(PATH) \
-                            POSTGRES_INCLUDE=${prefix}/include/postgresql14 \
-                            "POSTGRES_LIB=${prefix}/lib/postgresql14 -lssl"
+foreach s ${postgresql_suffixes} {
+    set p postgresql${s}
+    set i [lsearch -exact ${postgresql_ports} ${p}]
+    set conflicts [lreplace ${postgresql_ports} ${i} ${i}]
+    variant ${p} description "Build with postgresql${s} port" conflicts {*}${conflicts} "
+        depends_lib-append      port:${p}
+
+        configure.env           PATH=${prefix}/lib/${p}/bin:$env(PATH) \
+                                POSTGRES_INCLUDE=${prefix}/include/${p} \
+                                \"POSTGRES_LIB=${prefix}/lib/${p} -lssl\"
+    "
 }
 
 if {![variant_isset postgresql10] && \
     ![variant_isset postgresql11] && \
     ![variant_isset postgresql12] && \
     ![variant_isset postgresql13] && \
-    ![variant_isset postgresql14]} {
-        default_variants    +postgresql14
+    ![variant_isset postgresql14] && \
+    ![variant_isset postgresql15]} {
+        default_variants    +postgresql15
```

Comment:
it looks like this PR changes the default variant; in that case, one should increase the "revision".